### ISSUE TITLE
Python package names do not match with names declared in `jupyter-releaser` config

### ISSUE THREAD

**Issue by mahendrapaipuri** [jupyterlab-contrib/jupyterlab-topbar#117]
## Description

We declared Python package names in [`jupyter-releaser` config](https://github.com/jupyterlab-contrib/jupyterlab-topbar/blob/3744d6063adc32cc871c7274a6c21b996e2d8a04/package.json#L48-L50) with "underscores" while the names declared in [`pyproject.toml`](https://github.com/jupyterlab-contrib/jupyterlab-topbar/blob/3744d6063adc32cc871c7274a6c21b996e2d8a04/packages/logout-extension/pyproject.toml#L6) have "hypens".

When `jupyter-releaser` attempts to publish packages to PyPI, it checks if names declared in its config is [same as names in Wheel/Sdist](https://github.com/jupyter-server/jupyter_releaser/blob/81555cfd6a1708ad87130b822dab7e4891fd99cb/jupyter_releaser/lib.py#L430). This check will obviously fail and so skip publishing to PyPI.

## Reproduce

Try to publish to PyPI using `publish.yml` workflow.

## Expected behavior

Packages published to PyPI.

## Possible fix

Rename all the package names in `pyproject.toml` replacing hypens with underscores.

**Comment by jtpio** [comment]
> Rename all the package names in `pyproject.toml` replacing hypens with underscores.

Would replacing underscores by hyphens in the `package.json` also work?

**Comment by mahendrapaipuri** [comment]
> Would replacing underscores by hyphens in the package.json also work?

I dont think so as in `jupyter-releaser` [hypens are replaced by underscores](https://github.com/jupyter-server/jupyter_releaser/blob/81555cfd6a1708ad87130b822dab7e4891fd99cb/jupyter_releaser/lib.py#L399) before comparison is made

**Comment by jtpio** [comment]
OK thanks.

I guess we could then try that in a PR.

Normally the `check_release` workflow should publish the packages to a test PyPI server. Looking at a previous run (https://github.com/jupyterlab-contrib/jupyterlab-topbar/actions/runs/7990179602/job/21818416570), it looks like it was not doing that:

![image](https://github.com/jupyterlab-contrib/jupyterlab-topbar/assets/591645/c62d2cce-62de-4901-9daf-647291fd895b)

So we can check if it correctly publishes the packages on the test PR.

**Comment by mahendrapaipuri** [comment]
Yes, I can work on that.

Do you think it is worth raising this on `jupyter-releaser` to emit a warning when wheels and/or sdists found but skipped the publishing? There was no messages in the logs on why published has been skipped.

**Comment by jtpio** [comment]
> Do you think it is worth raising this on `jupyter-releaser` to emit a warning when wheels and/or sdists found but skipped the publishing? There was no messages in the logs on why published has been skipped.

That sounds useful yes. If I remember correctly there was also an issue when publishing on the voici packages (also using the `python_packages` of the releaser): https://github.com/voila-dashboards/voici/releases/tag/v0.6.0

**Comment by mahendrapaipuri** [comment]
Opened an [issue](https://github.com/jupyter-server/jupyter_releaser/issues/554) on `jupyter-releaser`

**Comment by jtpio** [comment]
Investigating in https://github.com/jupyterlab-contrib/jupyterlab-topbar/pull/119

**Comment by jtpio** [comment]
BTW @mahendrapaipuri I invited you to the repo with the proper rights to also be able to make releases with the releaser, in case you would like to help with the maintenance. Thanks!

**Comment by mahendrapaipuri** [comment]
Cheers @jtpio !! Sure, would be happy to give you a hand in maintenance!
